sdk/Sdk/Tasks/ZipDeploy/ZipDeploymentStatus.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Utilities;
+using Microsoft.NET.Sdk.Functions.Http;
+using Microsoft.NET.Sdk.Functions.Tasks;
+using Task = System.Threading.Tasks.Task;
+
+// IMPORTANT: Do not modify this file directly with major changes
+// This file is a copy from this project (with minor updates) -- https://github.com/Azure/azure-functions-vs-build-sdk/blob/b0e54a832a92119e00a2b1796258fcf88e0d6109/src/Microsoft.NET.Sdk.Functions.MSBuild/Microsoft.NET.Sdk.Functions.MSBuild.csproj
+// Please make any changes upstream first.
+
+namespace Microsoft.NET.Sdk.Functions.MSBuild.Tasks
+{
+    internal class ZipDeploymentStatus
+    {
+        private const int MaxMinutesToWait = 3;
+        private const int StatusRefreshDelaySeconds = 3;
+        private const int RetryCount = 3;
+        private const int RetryDelaySeconds = 1;
+
+        private readonly IHttpClient _client;
+        private readonly string _userAgent;
+        private readonly TaskLoggingHelper _log;
+        private readonly bool _logMessages;
+
+        public ZipDeploymentStatus(IHttpClient client, string userAgent, TaskLoggingHelper log, bool logMessages)
+        {
+            _client = client;
+            _userAgent = userAgent;
+            _log = log;
+            _logMessages = logMessages;
+        }
+
+        public async Task<DeployStatus> PollDeploymentStatusAsync(string deploymentUrl, string userName, string password)
+        {
+            DeployStatus deployStatus = DeployStatus.Pending;
+            var tokenSource = new CancellationTokenSource(TimeSpan.FromMinutes(MaxMinutesToWait));
+
+            if (_logMessages)
+            {
+                _log.LogMessage(StringMessages.DeploymentStatusPolling);
+            }
+            while (!tokenSource.IsCancellationRequested && deployStatus != DeployStatus.Success && deployStatus != DeployStatus.Failed && deployStatus != DeployStatus.Unknown)
```

Comment:
For me, at least, && implies a new line once the total expression is 'long'.  Would that help readability?

Reply:
+1